apps/voting/contracts/Voting.sol

```diff
@@ -213,6 +240,70 @@ contract Voting is IForwarder, AragonApp {
         _executeVote(_voteId);
     }
 
+    /**
+    * @notice Assign `_delegate` as the delegate for the sender
+    * @param _delegate address to delegate to
+    */
+    function setDelegate(address _delegate) external {
+        require(_delegate != address(0), ERROR_ZERO_ADDRESS_PASSED);
+        require(_delegate != msg.sender, ERROR_SELF_DELEGATE);
+
+        address prevDelegate = delegates[msg.sender].delegate;
+        require(_delegate != prevDelegate, ERROR_DELEGATE_SAME_AS_PREV);
+
+        if (prevDelegate != address(0)) {
+            _removeDelegatedAddressFor(prevDelegate, msg.sender);
+        }
+        _addDelegatedAddressFor(_delegate, msg.sender);
+    }
+
+    /**
+    * @notice Unassign `_delegate` from the sender
+    */
+    function resetDelegate() external {
+        address prevDelegate = delegates[msg.sender].delegate;
+        require(prevDelegate != address(0), ERROR_DELEGATE_NOT_SET);
+
+        _removeDelegatedAddressFor(prevDelegate, msg.sender);
+        delete delegates[msg.sender];
+    }
+
+    /**
+     * @notice Vote `_supports ? 'yes' : 'no'` in vote #`_voteId` on behalf of the `_voters` list
+     * @param _voteId Id for vote
+     * @param _supports Whether the delegate supports the vote
+     * @param _voters list of voters
+     */
+    function attemptVoteForMultiple(uint256 _voteId, bool _supports, address[] _voters) public voteExists(_voteId) {
+        Vote storage vote_ = votes[_voteId];
+        require(_isValidPhaseToVote(vote_, _supports), ERROR_CAN_NOT_VOTE);
+        bool hasManagedToVote = false;
+
+        address voter;
+        for (uint256 i = 0; i < _voters.length; ++i) {
+            voter = _voters[i];
+            require(_hasVotingPower(vote_, voter), ERROR_NO_VOTING_POWER);
+            if (_canVoteFor(vote_, msg.sender, voter)) {
+                _vote(_voteId, _supports, voter, true);
+                hasManagedToVote = true;
+            }
+        }
+
+        require(hasManagedToVote, ERROR_CAN_NOT_VOTE_FOR);
+    }
+
+    /**
+     * @notice Vote `_supports ? 'yes' : 'no'` in vote #`_voteId` on behalf of the `_voter`
+     * @param _voteId Id for vote
+     * @param _supports Whether the delegate supports the vote
+     * @param _voter address of the voter
+     */
+    function attemptVoteFor(uint256 _voteId, bool _supports, address _voter) external voteExists(_voteId) {
```

Comment:
`voter Exists` calls twice - not critical, but extra unnecessary gas spendings

Reply:
Fixed in 6c2840a9caea528321b3fbfbc0820c998c9aa37d.